mev_inspect/provider.py

```diff
@@ -1,9 +1,19 @@
 from web3 import AsyncHTTPProvider, Web3
 
+from mev_inspect.geth_poa_middleware import geth_poa_middleware
 from mev_inspect.retry import http_retry_with_backoff_request_middleware
+from mev_inspect.utils import RPCType
 
 
-def get_base_provider(rpc: str, request_timeout: int = 500) -> Web3.AsyncHTTPProvider:
+def get_base_provider(
+    rpc: str, request_timeout: int = 500, type: RPCType = RPCType.parity
+) -> Web3.AsyncHTTPProvider:
     base_provider = AsyncHTTPProvider(rpc, request_kwargs={"timeout": request_timeout})
-    base_provider.middlewares += (http_retry_with_backoff_request_middleware,)
+    if type is RPCType.geth:
+        base_provider.middlewares += (
```

Comment:
what is the advantage of adding middlewares at the provider, rather than using ```Web3.middleware_onion.add``` or ```Web3(middlewares=[geth_middleware])```?

Reply:
What do you propose, to set middlewares at MevInspector constructor? 
I'm not sure why it's set up like that, but for some reason when trying to set it up:
```
base_provider = AsyncHTTPProvider(rpc, request_kwargs={"timeout": request_timeout})
Web3(base_provider, modules={"eth": (AsyncEth,)}, middlewares=[geth_poa_middleware, http_retry_with_backoff_request_middleware,])
```
The function call
`w3.provider.middlewares `
returns empty set, so I'm assuming it has to be done at the provider.

Reply:
Hey, yeah it works when enabling debug "--http.api debug" on geth. With a full node I was able to start the listener at few days old blocks (had to update latest_block_update before running listener). 
Listener now infers rpc type from "trace_block" rpc call, but inspect and inspect-many still require the rpc_type parameter.